[✨feat] 일간 캘린더 API구현

jsoonworld opened the pull request:
# 📄 Work Description  
- `ScrapController`에 일간 캘린더 조회 API(`GET /api/v1/calendar/daily`)를 새롭게 추가했습니다.
- `ScrapService`에 `dailyScraps()` 메서드를 구현하여 사용자가 선택한 날짜 기준으로 스크랩 데이터를 조회할 수 있도록 했습니다.
- `ScrapRepositoryCustom`, `ScrapRepositoryImpl`에 사용자 ID와 날짜 기반의 스크랩 조회 쿼리를 새롭게 정의하고 구현했습니다.
- 기능 보장을 위해 `ScrapServiceTest`, `ScrapControllerTest`에 대한 단위 테스트도 함께 작성했습니다.

# 💭 Thoughts 
- 서비스 계층에선 `Clock` 주입을 통해 날짜 기반 로직이 테스트 가능한 구조가 되도록 했습니다. 실제 날짜에 의존하지 않게 하고자 기존 월간 조회 방식과 일관되게 처리했습니다.
- 날짜 파라미터 처리 시 `@DateTimeFormat`을 활용해 명확하게 포맷을 지정했으며, 형식 오류로 인한 예외를 방지하려고 했습니다.
- 컨트롤러와 서비스 사이에서 `DTO`를 일관성 있게 사용하면서도, `formattedDeadline`, `deadlineText`, `startYearMonth` 등 뷰에 필요한 데이터를 `DetailedScrap` 내에서 조합하도록 분리했습니다.
- '월간'과 '일간'이 일부 중복되는 로직을 가졌지만, 나중에 `ScrapResponseFactory` 같은 곳으로 분리할 수 있을지 함께 고민해보면 좋을 것 같아요!

# ✅ Testing Result  
<img width="423" alt="스크린샷 2025-06-08 오후 5 46 07" src="https://github.com/user-attachments/assets/39a95505-f39b-4c13-803b-f11cf6cffcfa" />
<img width="418" alt="스크린샷 2025-06-08 오후 5 46 27" src="https://github.com/user-attachments/assets/3a786423-a2bd-499b-b505-290a83045470" />


# 🗂 Related Issue  
- closed #98

leeeyubin committed:
feat(FilteringResponse) : FilteringResponse 구현

leeeyubin committed:
feat(FilterResponse) : Filter 네이밍 변경

# Conflicts:
#	src/main/kotlin/com/terning/server/kotlin/application/filter/FilterResponse.kt

leeeyubin committed:
feat(FilterController) : FilterController 구현

leeeyubin committed:
feat(FilterService) : FilterService 구현

leeeyubin committed:
feat(FilterControllerTest) : controller 테스트코드 작성

leeeyubin committed:
feat(FilterServiceTest) : service 테스트코드 작성

leeeyubin committed:
feat(FilterServiceTest) : ktlint 적용

jsoonworld committed:
feat(ScrapRepository) : 일간 스크랩 조회 메서드 정의

jsoonworld committed:
feat(ScrapRepositoryImpl) : 일간 스크랩 조회 쿼리 구현

jsoonworld committed:
feat(ScrapService) : 일간 스크랩 조회 기능 구현

jsoonworld committed:
feat(ScrapController) : 일간 스크랩 조회 API 추가

jsoonworld committed:
test(ScrapServiceTest) : 일간 스크랩 조회 기능 테스트 추가

jsoonworld committed:
test(ScrapServiceTest) : 일간 스크랩 조회 기능 테스트 추가

jsoonworld committed:
test(ScrapControllerTest) : 일간 스크랩 조회 API 테스트 추가

leeeyubin commented on src/main/kotlin/com/terning/server/kotlin/application/scrap/ScrapService.kt:
(사소..) 현재 해당 파일의 패키지가 `com.terning.server.kotlin.application.scrap`로 바뀌어야 할 것 같아요!

jsoonworld replied:
아이코! 그렇네요! 수정하겠습니다!

leeeyubin commented on src/test/kotlin/com/terning/server/kotlin/application/scrap/ScrapServiceTest.kt:
```diff
@@ -354,4 +354,73 @@ class ScrapServiceTest {
         }
     }
+
+    @Nested
+    @DisplayName("일간 스크랩 조회")
+    inner class DailyScrapTest {
+        @Test
+        @DisplayName("지정한 날짜에 해당하는 스크랩 정보를 반환한다")
+        fun returnsDailyScraps() {
```
(이것도 사소..) `ScrapControllerTest`처럼 여기도 given, when, then 주석으로 써주면 좋을 것 같아요!

jsoonworld replied:
적용해보겠습니다!

leeeyubin reviewed (approved):
달력 API 구현하시느라 고생많으셨어요!! 최고입니당

leeeyubin committed:
feat(FilterService) : Filter 에러 코드 적용

leeeyubin committed:
feat(FilterServiceTest) : Filter 테스트 코드 변경

leeeyubin committed:
feat(FilterService) : startDate 변수화

jsoonworld committed:
chore(format) : given, when, then 주석 적용

jsoonworld committed:
Merge branch 'feat/#95' into feat/#98

jsoonworld committed:
chore(merge) : 충돌 해결

jsoonworld committed:
chore(merge) : 충돌 해결

# Conflicts:
#	src/test/kotlin/com/terning/server/kotlin/application/scrap/ScrapServiceTest.kt

jsoonworld committed:
refactor(scrap) : 충돌 해결

jsoonworld committed:
chore(conflict) : 충돌 해결
